src/main/scala/is/hail/variant/GenomeReference.scala

```diff
@@ -0,0 +1,54 @@
+package is.hail.variant
+
+import is.hail.expr.JSONExtractGenomeReference
+import is.hail.utils._
+import org.json4s._
+import org.json4s.jackson.JsonMethods
+
+case class GenomeReference(name: String, contigs: Array[Contig], xContigs: Set[String],
+  yContigs: Set[String], mtContigs: Set[String], par: Array[Interval[Locus]]) extends Serializable {
+
+  def inX(contig: String): Boolean = xContigs.contains(contig)
+
+  def inY(contig: String): Boolean = yContigs.contains(contig)
+
+  def isMitochondrial(contig: String): Boolean = mtContigs.contains(contig)
+
+  def inXPar(locus: Locus): Boolean = inX(locus.contig) && par.exists(_.contains(locus))
+
+  def inYPar(locus: Locus): Boolean = inY(locus.contig) && par.exists(_.contains(locus))
+}
+
+object GenomeReference {
+  def GRCh37 = fromResource("reference/human_g1k_v37.json")
+
+  def fromResource(file: String): GenomeReference = {
+    val resourceStream = Thread.currentThread().getContextClassLoader.getResourceAsStream(file)
+
+    try {
+      if (resourceStream == null) {
+        throw new RuntimeException(s"Could not read genome reference file `$file'.")
+      }
+
+      val json = JsonMethods.parse(resourceStream)
+
+      json.extract[JSONExtractGenomeReference].toGenomeReference
```

Comment:
Much cleaner!  Very nice.  Digging into the json4s code, looks like we define custom extractors (for things like Interval) which we should use instead of these JSONExtract classes I created.  Will be cleaner again, but we can leave that for another PR.